lavalink/models.py

```diff
@@ -228,14 +228,18 @@ def add(self, requester: int, track: dict, extra: dict = None, index: int = None
                 track = self.queue.pop(0)
 
         self.current = track
+        options = {}
+
+        if 0 > end_time > track.duration or 0 > start_time > track.duration:
+            raise ValueError("end_time or start_time is either less than 0 or greater than the track's duration")
+        elif end_time == 0:
```

Comment:
It doesn't make sense to set a value to end_time here.

Reply:
Moreover this whole change is kindda weird, it doesn't allow to have noreplace, endtime and/or start time at the same time ? You're not checking if start time is valid as well. See my old example if it can help you

Reply:
Oh oops, they should've been separate if statements, will fix